pkg/controller/networkpolicy/antreanetworkpolicy.go

```diff
@@ -204,3 +217,48 @@ func (n *NetworkPolicyController) processAntreaNetworkPolicy(np *crdv1alpha1.Net
 	}
 	return internalNetworkPolicy
 }
+
+func (n *NetworkPolicyController) processAppliedTo(namespace string, appliedTo []crdv1alpha1.NetworkPolicyPeer, appliedToGroupNamesSet sets.String) ([]string, error) {
+	var appliedToGroupNames []string
+	for _, at := range appliedTo {
+		var atg string
+		if at.Group != "" {
+			var err error
+			atg, err = n.processAppliedToGroupForNamespacedGroup(namespace, at.Group)
+			if err != nil {
+				return appliedToGroupNames, err
+			}
+		} else {
+			atg = n.createAppliedToGroup(namespace, at.PodSelector, at.NamespaceSelector, at.ExternalEntitySelector)
+		}
+		if atg != "" {
+			appliedToGroupNames = append(appliedToGroupNames, atg)
+			appliedToGroupNamesSet.Insert(atg)
+		}
+	}
+	return appliedToGroupNames, nil
+}
+
+func (n *NetworkPolicyController) processAppliedToGroupForNamespacedGroup(namespace, groupName string) (string, error) {
+	// Retrieve Group for corresponding entry in the AppliedToGroup.
+	g, err := n.grpLister.Groups(namespace).Get(groupName)
+	if err != nil {
+		// The Group referred to has not been created yet.
+		return "", nil
```

Comment:
We removed the requirement that Groups/ClusterGroups should be a strong reference a while ago. Most K8s resources are designed this way: consider Role and RoleBinding. When you specify a Role in a RoleBinding which does not exist yet, the apiserver will not throw an error at you; instead the resource you created will simply have no effect since it failed to reconcile

Reply:
Sounds good, thanks.